owncloudApp/src/main/java/com/owncloud/android/MainApp.java

```diff
@@ -264,18 +233,25 @@ public static String getUserAgent() {
         return String.format(appString, version);
     }
 
-    public static boolean isBeta() {
-        boolean isBeta = false;
-        try {
-            String packageName = getAppContext().getPackageName();
-            PackageInfo packageInfo = getAppContext().getPackageManager().getPackageInfo(packageName, 0);
-            String versionName = packageInfo.versionName;
-            if (versionName.contains(BETA_VERSION)) {
-                isBeta = true;
-            }
-        } catch (PackageManager.NameNotFoundException e) {
-            e.printStackTrace();
+    public static boolean isDeveloper() {
+        return mDeveloper;
+    }
+
+    public void readIsDeveloper() {
+        mDeveloper = BuildConfig.DEBUG || PreferenceManager.getDefaultSharedPreferences(getApplicationContext())
+                .getInt(CLICK_DEV_MENU, 0) > CLICKS_NEEDED_TO_BE_DEVELOPER;
+
+
+        if (isDeveloper()) {
```

Comment:
I would not include the code to start logging in here since this method (readIsDeveloper) is not expected to do that. Instead, return a boolean and call a different method to start logging from onCreate().

Reply:
This is my tested way, I will not change this anymore and I don't support your suggestion.
If you want to have it change, I pushed branch to 'upstream' remote, so you can change it and rebase ....